dowhy/gcm/validation.py

```diff
@@ -0,0 +1,157 @@
+"""Contains a method to reject the causal graph and validate causal causal mechanisms such as post non-linear models.
+"""
+from enum import Enum, auto
+from typing import Any, Callable, Dict, Tuple, Union, Optional, Set
+
+import networkx as nx
+import numpy as np
+import pandas as pd
+from statsmodels.stats.multitest import multipletests
+
+from dowhy.gcm.cms import InvertibleStructuralCausalModel
+from dowhy.gcm.graph import DirectedGraph, get_ordered_predecessors, validate_causal_graph, is_root_node
+from dowhy.gcm.independence_test import kernel_based
+
+
+class RejectionResult(Enum):
+    REJECTED = auto()
+    NOT_REJECTED = auto(),
+
+
+def reject_causal_structure(causal_graph: DirectedGraph,
+                            data: pd.DataFrame,
+                            independence_test: Callable[[np.ndarray, np.ndarray], float] = kernel_based,
+                            conditional_independence_test: Callable[
+                                [np.ndarray, np.ndarray, np.ndarray], float]
+                            = kernel_based,
+                            significance_level: float = 0.05,
+                            fdr_control_method: Optional[str] = 'fdr_bh') \
+        -> Tuple[RejectionResult, Dict[str, Dict[str, Dict[str, Union[bool, float, Dict[str, Union[bool, float]]]]]]]:
+    """Validates the assumptions in a causal graph against data. To this end, at each node, we test if the node is dependent on each of its parents, and test the local Markov condition.
```

Comment:
If I remember correctly, the local markov property implies the global property, correct?
If so, it will be good to state that here.

Reply:
Yes good point, added it as a remark to the docstring.